### ISSUE TITLE
Make no-unused-labels message more helpful in arrow function blocks

### ISSUE THREAD

**Issue by btmills** [eslint/eslint#8613]
**Tell us about your environment**

* **ESLint Version:** Demo
* **Node Version:** n/a
* **npm Version:** n/a

**What parser (default, Babel-ESLint, etc.) are you using?**

Default (demo)

**Please show your full configuration:**

Change ECMA Version to 6, leave rest as default

```
{
    parserOptions: {
        ecmaVersion: 6
    }
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
/* eslint no-unused-labels: 2 */
[].map(thing => { // Missing parens around block body
    value: thing.value * 2
});
```

**What did you expect to happen?**

Inspired by http://stackoverflow.com/q/43993719/4510592, it would be nice if `no-unused-labels` could suggest "Did you mean to return an object? Wrap the arrow body in parentheses." (wording up for bikeshedding).

**What actually happened? Please include the actual, raw output from ESLint.**

```
2:5 - 'value:' is defined but never used. (no-unused-labels)
```

**Comment by not-an-aardvark** [comment]
This would also apply to cases like

```js
return
{
    ok: true
};
```

**Comment by mysticatea** [comment]
In this case, [no-labels] and [no-unused-expressions] also report the code. In @not-an-aardvark's case, [no-lone-blocks] and [no-unreachable] are as well. Should we change the message of those rules? Or maybe new rule like `no-blocks-like-objects` should be.

[no-labels]:http://eslint.org/docs/rules/no-labels
[no-unused-expressions]: http://eslint.org/docs/rules/no-unused-expressions
[no-lone-blocks]: http://eslint.org/docs/rules/no-lone-blocks
[no-unreachable]: http://eslint.org/docs/rules/no-unreachable

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get enough support from the team and so I'm closing it. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to reach consensus after a long time tend to never do it, and as such, we close those issues. This doesn't mean the idea isn't interesting, just that it's not something the team can commit to.
